provider/source/src/time_zones/convert.rs

```diff
@@ -96,9 +98,79 @@ fn compute_meta_zone_ids_btreemap(
     for (meta_zone_id, meta_zone_id_data) in meta_zone_ids_resource.iter() {
         meta_zone_ids.insert(meta_zone_id_data.long_id.to_string(), *meta_zone_id);
     }
+    // TODO(#1781): Remove this special case once the short id is updated in CLDR
+    meta_zone_ids.insert("Yukon".to_owned(), MetazoneId(tinystr!(4, "yuko")));
     meta_zone_ids
 }
 
+type Offsets = (EightsOfHourOffset, Option<EightsOfHourOffset>);
+
+// TODO: load this from a better source. Ideally this data would be in CLDR
+fn offsets_for_meta(meta_zones_territory: &Vec<MetazoneTerritory>, meta: &str) -> Option<Offsets> {
+    let (mut standard, mut daylight) = (None, None);
+    let mut first_zone = "";
+    for m in meta_zones_territory {
+        if m.map_zone.other != meta {
+            continue;
+        }
+        if m.map_zone.zone_type == "Europe/Minsk" {
+            // Minsk has an incorrect metazone, so ignore it
+            // https://unicode-org.atlassian.net/browse/CLDR-17927
+            continue;
+        }
+        let Some((new_standard, new_daylight)) = offsets_for_iana(&m.map_zone.zone_type) else {
+            continue;
+        };
+        if let Some(standard) = standard {
+            match (standard, daylight, new_standard, new_daylight) {
+                // Same standard times, no DST, allowed
+                (s1, None, s2, None) if s1 == s2 => {}
+                // Same standard times, one DST, allowed
+                (s1, Some(_), s2, None) | (s1, None, s2, Some(_)) if s1 == s2 => {}
+                // Same standard times, same DST times, allowed
+                (s1, Some(d1), s2, Some(d2)) if s1 == s2 && d1 == d2 => {}
+                // One tz has DST, and it's the other's standard time, sketchy but allowed
+                (s, None, _, Some(d)) | (_, Some(d), s, None) if s == d => {}
+                // Both tzs have DST but swapped, sketchy but allowed
+                (s1, Some(d1), s2, Some(d2)) if s1 == d2 && d1 == s2 => {}
```

Comment:
Question: do these "sketchy" cases, especially the second one, happen in data?

Reply:
Yes, for example Ireland has DST in the winter for some reason.

Reply:
Ok, are you testing that the Ireland time zone display names work out correctly?

Reply:
There's an Ireland test in the fixtures. Before it had to consider the weirdness, now it just looks like any other test.